### ISSUE TITLE
gs and Time series split

### ISSUE THREAD

**Issue by franxav79** [skorch-dev/skorch#924]
I have the same issue as #837 train_valid, GridSearchCV and TimeSeriesSplit, how I can recover the valid_loss from gs and see how well the model train, is possible?

Or the aproach is to take the best parameters result just with train_loss and to test in validation set by separate.
I am not using call backs.

net = NeuralNet(
module=LSTM_E_1, max_epochs=epochs,
optimizer=torch.optim.Adam,
criterion=criterion,
lr=lr, verbose=1,
device=device,
)

params = {
'lr': [1, 0.1, 0.01, 0.001, 0.0001]}

net.set_params(verbose=0, train_split= None)
tscv = TimeSeriesSplit(n_splits=6, test_size=2)
gs = GridSearchCV(net,
params,
refit=True,
scoring='r2',
verbose=1, cv=tscv,
#return_valid_score=True
)

**Comment by BenjaminBossan** [comment]
> how I can recover the valid_loss from gs and see how well the model train, is possible?

Maybe I misunderstand, but after fitting the grid search, you can access the `cv_results_` attribute to recover the validation loss. E.g., after your code:

```python
gs.fit(X, y)
print(gs.cv_results_)
# if you have pandas installed, you can also use this, which is prettier:
print(pd.DataFrame(gs.cv_results_))
```

Note that the `cv_results_` are the cross validation results _from sklearn grid search_, not from skorch. Therefore, the values you see there will differ from what skorch prints in its own logs.

Generally, since grid search already validates the scores for you, you should _turn off skorch validation_:

`net = NeuralNet(..., train_split=False)`

(You may also want to set `verbose=0` if there is too much output printed.)

The reason why in the initial issue that you linked it was not so easy is because the user wanted to use `EarlyStopping`, which by default requires the skoch validation. But since you don't have those requirements, you should be fine.

**Comment by franxav79** [comment]
Tks for the help

**Comment by BenjaminBossan** [comment]
Hopefully, this solves your issue. If something new comes up, feel free to re-open the issue.
